src/Transports.AspNetCore/WebSockets/BaseSubscriptionServer.cs

```
@@ -0,0 +1,488 @@
+        {
+            await ErrorIdCannotBeBlankAsync(message);
+            return;
+        }
+        var messageId = message.Id!;
+
+        var dummyDisposer = new DummyDisposer();
+
+        try
+        {
+            if (overwrite)
+            {
+                Subscriptions[messageId] = dummyDisposer;
+            }
+            else
+            {
+                if (!Subscriptions.TryAdd(messageId, dummyDisposer))
+                {
+                    await ErrorIdAlreadyExistsAsync(message);
+                    return;
+                }
+            }
+
+            var result = await ExecuteRequestAsync(message);
+            CancellationToken.ThrowIfCancellationRequested();
+            if (!Subscriptions.Contains(messageId, dummyDisposer))
+                return;
+            if (result.Streams?.Count == 1)
+            {
+                // do not return a result, but set up a subscription
+                var stream = result.Streams!.Single().Value;
+                // note that this may immediately trigger some notifications
+                var disposer = stream.Subscribe(new Observer(this, messageId, _options.WebSockets.DisconnectAfterErrorEvent, _options.WebSockets.DisconnectAfterAnyError));
+                try
+                {
+                    if (Subscriptions.CompareExchange(messageId, dummyDisposer, disposer))
+                    {
+                        disposer = null;
+                    }
+                }
+                finally
+                {
+                    disposer?.Dispose();
+                }
+            }
+            else if (result.Executed)
+            {
+                await SendSingleResultAsync(message, result);
+            }
+            else
+            {
+                await SendErrorResultAsync(message, result);
+            }
+        }
+        catch (OperationCanceledException) when (CancellationToken.IsCancellationRequested)
+        {
+            throw;
```

In `WebSocketConnection` exception is swallowed:
```cs
 catch (WebSocketException) when (RequestAborted.IsCancellationRequested)
        {
        }
```

Why do you throw here?

---

I fail to understand; OCE should always throw, and does so in WebSocketConnection as well.

Are you suggestion I should add this?

```cs
catch (WebSocketException)
{
    throw; // handled in WebSocketConnection
}
```

---

> Are you suggesting I should add this?

No. My question was about throw or not here when cathing OCE .